### PR TITLE
Mhv 46752 hi fi mock updates

### PR DESCRIPTION
**Pull request by robertroddev**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Summary

Made some UI changes, added a sort button, and breadcrumbs to medications list view.

## Related issue(s)

[Update Medications List page based on Hi-Fi mocks](https://jira.devops.va.gov/browse/MHV-46752)

## Testing done

no unit testing done, will do testing on a separate pr.

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  |image|image|
| Desktop |image|image|

## What areas of the site does it impact?

my-health/Medications 

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/mhv/medications/components/MedicationsList/MedicationsListCard.jsx [hunk 1]
```diff
@@ -110,27 +110,26 @@ const MedicationsListCard = props => {
...
+            {rx.prescriptionName}
           </Link>
-        </div>
+        </h4>
+        {rx.dispensedDate ? (
+          <div>
+            Last filled on {dateFormat(rx.dispensedDate, 'MMMM D, YYYY')}
+          </div>
+        ) : (
+          <div>You haven’t filled this prescripton yet.</div>
```
Comment (acrollet) [inline src/applications/mhv/medications/components/MedicationsList/MedicationsListCard.jsx:129]:
```suggestion
          <div>You haven’t filled this prescription yet.</div>
```

File: src/applications/mhv/medications/containers/Prescriptions.jsx [hunk 2]
```diff
@@ -112,15 +114,21 @@ const Prescriptions = () => {
...
           url: '/my-health/medications/prescriptions/',
           label: 'Medications',
-        }),
-      );
-    }
-  });
+        },
+      ),
+    );
+  }, []);
```
Comment (acrollet) [inline src/applications/mhv/medications/containers/Prescriptions.jsx:131]:
```suggestion
  }, [dispatch]);
```

Comment (robertbylight) [reply src/applications/mhv/medications/containers/Prescriptions.jsx]: Thanks I just pushed this change

File: src/applications/mhv/medications/components/MedicationsList/MedicationsListCard.jsx (hunk 1 above)
Comment (robertbylight) [reply src/applications/mhv/medications/components/MedicationsList/MedicationsListCard.jsx]: Thanks I just pushed this change

**Review by KolbyVA — APPROVED** [review]

File: src/applications/mhv/medications/containers/Prescriptions.jsx [hunk 3]
```diff
@@ -27,13 +27,15 @@ const Prescriptions = () => {
...
     () => {
-      const newList = [...prescriptions];
-      newList.sort(a => {
-        return a.refillStatus.toLowerCase() === sortOption.toLowerCase()
-          ? -1
-          : 0;
-      });
-      dispatch(setSortedRxList(newList));
+      if (sortOption) {
+        const newList = [...prescriptions];
```
Comment (acrollet) [inline src/applications/mhv/medications/containers/Prescriptions.jsx:31]: I know unit tests will go in a separate PR, it would be good to split this logic into a separate function and get it unit tested there :)

**Review by acrollet — APPROVED** [review]

File: src/applications/mhv/medications/containers/Prescriptions.jsx (hunk 3 above)
Comment (robertbylight) [reply src/applications/mhv/medications/containers/Prescriptions.jsx:31]: aah Im sorry Adrian I just merged it and missed this comment.

Comment (acrollet) [reply src/applications/mhv/medications/containers/Prescriptions.jsx:31]: @robertbylight nope all good, I just meant to handle in the testing PR 🙂
